src/coins.cpp

```diff
@@ -1430,16 +1428,20 @@ CValidationState::Code CCoinsViewCache::IsScTxApplicableToState(const CTransacti
                 __func__, __LINE__, tx.ToString(), csw.ToString());
             return CValidationState::Code::INVALID;
         }
-        
-        int32_t idx = csw.actCertDataIdx;
-        // index should have been already checked at this point, at() throws an exception
-        CFieldElement certDataHash = tx.GetVActCertData().at(idx);
-        if (GetActiveCertDataHash(csw.scId) != certDataHash)
+
+        if (GetActiveCertDataHash(csw.scId) != csw.actCertData)
```

Comment:
Let's add the comment here, that it's fine to have `actCertData == CFieldElement()`.
Honestly, it was confusing before I saw an additional valid case check for `!csw.actCertData.IsNull()` in `checkTxSemanticValidity` and rpc.

Reply:
Done